code/datums/components/attachment_handler.dm

```diff
@@ -0,0 +1,304 @@
+/datum/component/attachment_handler
+	///Assoc list that stores the refs of the attachments to the parent. 'slot = reference'
+	var/list/slots = list()
+	///Typepath list that stores type data of allowed attachments. The attachment needs the attachment element to be able to attach regardless of what is stored here.
+	var/list/attachables_allowed = list()
+	///Proc the parent calls on attach.
+	var/datum/callback/on_attach
+	///Proc the parent calls on detach.
+	var/datum/callback/on_detach
+	///List of offsets for the parent that adjusts the attachment overlay. slot1_x = 1, slot1_y = 1, slot2_x = 3, slot2_y = 7, etc.
+	var/list/attachment_offsets = list()
+	///List of the attachment overlay images. This is so that we can easily swap overlays in and out.
+	var/list/attachable_overlays = list()
+	///List of the icon states for the attachable_overlays. This exists so the parent or the attachment can change the overlay icon state.
+	var/list/overlay_icon_states = list()
+
+
+/datum/component/attachment_handler/Initialize(_slots, list/_attachables_allowed, list/_attachment_offsets, list/starting_attachmments, datum/callback/_on_attach, datum/callback/_on_detach, list/overlays = list())
```

Comment:
use the normal name and use ie. src.slots = thing, don't use _name, its on contribution guidelines btw